### PR TITLE
Adding testing, fixing discovered bugs and compiler warnings

### LINKED ISSUE #10: Limitation to images which size along each dimension has only 2,3, and 5 as prime factors
When using `itkPhaseCorrelationImageRegistration`, for example with the example that is implemented, one is restricted to process images which size along each dimension has only 2,3, and 5 as prime factors. This is the case when using the VNL implementation of the FFT transform in ITK, and not the case when using the FFTW implementation, but the latter has license restriction. The filter should automatically pad the input, or preprocess the image such that the user can input image of any size.

### LINKED ISSUE #16: PhaseCorrelationImageRegistrationMethod VerifyInputInformation checks Spacing and Direction
FixedImage and MovingImage should have the same Spacing and Direction.

### PR DESCRIPTION
**Pull request by dzenanz**
Based on, and supersedes https://github.com/InsightSoftwareConsortium/ITKMontage/pull/15
Closes #16.
Closes #10.

### CODE REVIEW

**Comment by dzenanz** [comment]
The failure is due to correctly failing `*Files` test. Should we merge as-is or comment out the failing test?

**Comment by thewtex** [comment]
@dzenanz Can we add the `WILL_FAIL` property to the test?:

  https://cmake.org/cmake/help/v3.0/prop_test/WILL_FAIL.html

A good name could be `*SameSpacing` instead of `*Files`.

File: itk-module.cmake [hunk 1]
```diff
@@ -17,6 +17,7 @@ itk_module(Montage
     ITKFFT
     ITKTransform
   TEST_DEPENDS
+    ITKIOTransformInsightLegacy
```
Comment (thewtex) [inline itk-module.cmake]: HDF5 transform is preferred, but that can be done later.

**Comment by thewtex** [comment]
> @dzenanz Can we add the WILL_FAIL property to the test?:
> A good name could be *SameSpacing instead of *Files.

Sorry, looking closer, that suggestion was not correct.


A better approach is to increase the tolerance until we implement a method with higher precision, e.g. parabolic interpolation.

**Pull request merged**